app/models/pieces.py

```diff
@@ -31,7 +32,28 @@ class Rook(Piece): ...
 class Queen(Piece): ...
 
 
-class King(Piece): ...
+class King(Piece):
+    def validate_move(self, board: np.array, move: PieceMove):
+        super().validate_move(board, move)
+
+        # Restrict the king's movement to one block
+        if abs(move.from_pos[0] - move.to_pos[0]) <= 1 and abs(move.from_pos[1] - move.to_pos[1]) <= 1:
+            # Check if the move is in the same column
+            if move.from_pos[0] == move.to_pos[0]:
```

Comment:
There is the scenario where the "from" and the "to" position are the same.
In this case, we should raise a wrong move.

Reply:
nice one! resolved!